omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb

```diff
@@ -49,36 +49,45 @@
   end
 end
 
-ssl_keyfile = File.join(nginx_ca_dir, "#{node['private_chef']['nginx']['server_name']}.key")
-ssl_crtfile = File.join(nginx_ca_dir, "#{node['private_chef']['nginx']['server_name']}.crt")
-ssl_dhparam = File.join(nginx_ca_dir, 'dhparams.pem')
-
-# Generate self-signed SSL certificate
-openssl_x509 ssl_crtfile do
-  common_name node['private_chef']['nginx']['server_name']
-  org node['private_chef']['nginx']['ssl_company_name']
-  org_unit node['private_chef']['nginx']['ssl_organizational_unit_name']
-  country node['private_chef']['nginx']['ssl_country_name']
-  key_length node['private_chef']['nginx']['ssl_key_length']
-  expire node['private_chef']['nginx']['ssl_duration']
-  owner 'root'
-  group 'root'
-  mode '0644'
-end
 
-# Generate dhparam.pem for better secrecy
-openssl_dhparam ssl_dhparam do
-  key_length node['private_chef']['nginx']['dhparam_key_length']
-  generator node['private_chef']['nginx']['dhparam_generator_id']
-  owner 'root'
-  group 'root'
-  mode '0644'
+
+# Generate self-signed SSL certificate unless the user has provided one
+if (node['private_chef']['nginx']['ssl_certificate'].nil? &&
+    node['private_chef']['nginx']['ssl_certificate_key'].nil?)
+
+   ssl_keyfile = File.join(nginx_ca_dir, "#{node['private_chef']['nginx']['server_name']}.key")
+   ssl_crtfile = File.join(nginx_ca_dir, "#{node['private_chef']['nginx']['server_name']}.crt")
+
+   openssl_x509 ssl_crtfile do
+     common_name node['private_chef']['nginx']['server_name']
```

Comment:
If this value changes, the user would have to delete the generated cert and key files and reconfigure to have a matching cert regenerated. Do we need to catch this edge case?